examples/repl.rs

```diff
@@ -142,71 +209,101 @@ mod example {
+            // Test that `a_function` resolves to a function.
+            let breakpoint_addr = context.debuginfo().get_symbol_address("breakpoint").unwrap() + 4 /* prologue */;
+            // Write breakpoint to the `breakpoint` function.
+            let mut pause_inst = 0 as libc::c_ulong;
+            unsafe {
+                context
+                    .remote()?
+                    .read()
+                    .read(&mut pause_inst, breakpoint_addr)
+                    .apply()
+                    .unwrap();
             }
-            Some("exec") => {
-                if let Some(cmd) = parts.next() {
-                    println!("Starting program: {}", cmd);
-                    let (remote, status) = LinuxTarget::launch(cmd)?;
-                    println!("{:?}", status);
-                    context.set_remote(remote);
-                }
+            // pause (rep nop); ...
+            assert_eq!(
+                &pause_inst.to_ne_bytes()[0..2],
+                &[0xf3, 0x90],
+                "Pause instruction not found"
+            );
+            let mut breakpoint_inst = pause_inst.to_ne_bytes();
+            // int3; nop; ...
+            breakpoint_inst[0] = 0xcc;
+            nix::sys::ptrace::write(
+                context.remote()?.pid(),
+                breakpoint_addr as *mut _,
+                libc::c_ulong::from_ne_bytes(breakpoint_inst) as *mut _,
+            )
+            .unwrap();
+
+            return Ok(());
+        }
+
+        let command = ReplCommand::from_str(command)?;
+        match command {
+            ReplCommand::Help(()) => {
+                ReplCommand::print_help(std::io::stdout(), color.get()).unwrap();
             }
-            Some("attach") => {
-                if let Some(pid) = parts.next() {
-                    let pid = nix::unistd::Pid::from_raw(pid.parse()?);
-                    println!("Attaching to process {}", pid);
-                    let (remote, status) = LinuxTarget::attach(
-                        pid,
-                        AttachOptions {
-                            kill_on_exit: false,
-                        },
-                    )?;
-                    println!("{:?}", status);
-                    // FIXME detach or kill old remote
-                    context.set_remote(remote);
-                }
+            ReplCommand::Exec(cmd) => {
+                println!("Starting program: {}", cmd.display());
+                let (remote, status) = LinuxTarget::launch(&cmd)?;
+                println!("{:?}", status);
+                context.set_remote(remote);
             }
-            Some("detach") => {
+            ReplCommand::Attach(pid) => {
+                let pid = nix::unistd::Pid::from_raw(pid.parse()?);
+                println!("Attaching to process {}", pid);
+                let (remote, status) = LinuxTarget::attach(
+                    pid,
+                    AttachOptions {
+                        kill_on_exit: false,
```

Comment:
While we are at it, it might be good to move command implementation code, longer than a line into functions of their own ?